### PR TITLE
#2: Add project skeleton

### LINKED ISSUE #2: Add project skeleton with required file and directory structure
In this issue we are creating the project skeleton for the necessary files and directories.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #2.

This PR add skeleton directories and files for the mandatory documenation of OFT sub-projects.

### CODE REVIEW

File: doc/developer_guide.md
Comment (kaklakariada) [inline doc/developer_guide.md]:
```suggestion
By default, the AsciiDoc plugin is built with Java 17. To build and test with a later version of Java, add argument `-Djava.version=21` to the Maven command.
```

File: doc/developer_guide.md
Comment (kaklakariada) [inline doc/developer_guide.md]:
```suggestion
mvn -Djava.version=21 package -DskipTests
```

File: doc/spec/system_requirement.md
Comment (kaklakariada) [inline doc/spec/system_requirement.md:1]: Should be in asciidoc format to demonstrate the importer.

File: doc/spec/design.md
Comment (kaklakariada) [inline doc/spec/design.md:1]: Should be in asciidoc format to demonstrate the importer.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — DISMISSED** [review]

**Comment by sophokles73** [comment]
@redcatbear seems like there are some broken links that need to be fixed in order for this to get merged?

**Comment by kaklakariada** [comment]
@sophokles73 The broken links are expected, they should be fixed after we published the first release to maven central. This PR is just waiting for a final review by @redcatbear , because I also added some changes.

You can already start development based on `main` or this branch, there should not be too many merge conflicts.

**Review by redcatbear — COMMENTED** [review]
I can only comment (not approve), since I am the original author. @kaklakariada, your changes look good to me. Broken links expected, as you already pointed out. Until the first Maven release.

@sophokles73 , you can comment the links until then if you like in your branch.

**Review by kaklakariada — APPROVED** [review]

**Comment by kaklakariada** [comment]
I excluded the link check from required pr checks for now, so it won't block merging.

**Pull request merged**
